Sources/SimultaneouslyScrollView/Extensions/MulticastScrollViewDelegate.swift

```diff
@@ -0,0 +1,107 @@
+import UIKit;
+
+class MulticastScrollViewDelegate: NSObject, UIScrollViewDelegate {
+    // Store weak references to delegates
+    private var delegates = NSHashTable<UIScrollViewDelegate>.weakObjects()
+
+    func addDelegate(_ delegate: UIScrollViewDelegate) {
+        delegates.add(delegate)
+    }
+
+    func removeDelegate(_ delegate: UIScrollViewDelegate) {
+        delegates.remove(delegate)
+    }
+
+    func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        for delegate in delegates.allObjects {
+            delegate.scrollViewDidScroll?(scrollView)
+        }
```

Comment:
Nit: this `for` loops produce a lot of code duplications. What do you think of introducing a convenience method to abstract this away? 🤔 

e.g. something like this:

```swift
    private func callAll(_ block: (_ delegate: UIScrollViewDelegate) -> Void) {
        delegates.allObjects.forEach(block)
    }

    // usage

    callAll { $0.scrollViewDidScroll?(scrollView) }
```

Reply:
I also thought about that. But then I thought `callAll` is essentially `forEach`, just saving the `delegates.allObjects`... and it breaks down when there's a return value. So I kinda preferred the slightly repetitive version, but I'm fine with `callAll`, too.